kbcstorage/workspaces.py

```diff
@@ -0,0 +1,112 @@
+"""
+Manages workspace requests to the API.
+
+Full documentation `here`.
+
+.. _here:
+    http://docs.keboola.apiary.io/#reference/workspaces/
+"""
+
+from kbcstorage.base import Endpoint
+
+
+def _make_body(mapping):
+    """
+    Given a dict mapping Keboola tables to aliases, construct the body of
+    the HTTP request to load said tables.
+
+    Args:
+        mapping(:obj:`dict`): Keys contain the full names of the tables to
+            be loaded (ie. 'in.c-bucker.table_name') and values contain the
+            aliases to which they will be loaded (ie. 'table_name').
+        preserve(bool): If True, does not clear the workspace of existing
+            tables. If False, clears the workspace of tables before loading.
+            Default False.
+    """
+    body = {}
+    template = 'input[{0}][{1}]'
+    for i, (k, v) in enumerate(mapping.items()):
+        body[template.format(i, 'source')] = k
+        body[template.format(i, 'destination')] = v
+
+    return body
+
+
+class Workspaces(Endpoint):
+    """
+    Workspaces Endpoint
+    """
+    def __init__(self, url, token):
+        """
```

Comment:
@odinuv so instead of passing the token to each endpoint you'd pass the client class, I don't see any benefit in that. 
> .. it's just that this way one has to pass the token all around the code ...

Not really, you would only pass the token in the `Client.__init__()` method, which would internally distribute it to the endpoints.

i.e
```python
from urllib.parse import urljoin

class StorageClient:
    def __init__(self, token):
        self.base_url = 'https://connection.keboola.com/v2/storage/'
        workspaces_url = self.base_url
        buckets_url = urljoin(self.base_url, 'buckets')

        self.workspaces = Workspaces(workspaces_url, token)
        self.buckets = Buckets(buckets_url, token)
```

and use like this

```shell
>>> client = StorageClient('token123')
>>> client.workspaces.list()
{
    "id": 234,
    "name": "boring_wozniak",
    "component": "wr-db",
    "configurationId": "aws-1"
   # ... omitted for clarity
}
>>> client.buckets.all()
{ response listing all buckets }

```

I think this has several benefits, the client class is essentially your gateway to the world of the keboola storage api:
- you don't pass anything anywhere (tokens, clients,...)
- you don't have to import and instantiate the endpoints individually
- All endpoints are in one place, convinient and ready to use

---
Now when I am thinking about it, what would the `Client` class do in your example? To me it looks just like overly complicated conatiner for the `base_url` and `token`

```python
class Client:
    def __init__(self, base_url, token):
        self.base_url = base_url
        self.token = token
```